src/utils/__tests__/awsLambdaLayerManager.test.ts

```diff
@@ -0,0 +1,90 @@
+import { DescribeRegionsCommandOutput } from '@aws-sdk/client-ec2';
+import * as awsManager from '../awsLambdaLayerManager';
+
+const CANONICAL_SEPARATOR = ':';
+
+const COMPATIBLE_RUNTIME_DATA = {
+  name: 'test runtime',
+  versions: ['test version 1', 'test version 2'],
+};
+const AWS_TEST_REGIONS = ['test aws region 1', 'test aws region 2'];
+
+/** The default region used to fetch all available AWS regions. */
+const DEFAULT_REGION = 'us-east-2';
+
+function getTestAwsLambdaLayerManager(): awsManager.AwsLambdaLayerManager {
+  return new awsManager.AwsLambdaLayerManager(
+    COMPATIBLE_RUNTIME_DATA,
+    'test layer name',
+    'test license',
+    Buffer.alloc(0),
+    AWS_TEST_REGIONS
+  );
+}
+
+describe('canonical', () => {
+  test('get canonical name', () => {
+    const manager = getTestAwsLambdaLayerManager();
+    const canonicalSuffix = manager
+      .getCanonicalName()
+      .split(CANONICAL_SEPARATOR)[1];
+    expect(canonicalSuffix).toBe(COMPATIBLE_RUNTIME_DATA.name);
```

Comment:
nit/personal opinion: replacing `COMPATIBLE_RUNTIME_DATA.name` with the actual value (`test runtime`) would make the test a bit more robust because it'd avoid failing in unfortunate situations when e.g. `name` attribute is changed and also `getCanonicalName` changes the logic, in that situation the test has a chance of passing.